dora/src/module.rs

```diff
@@ -0,0 +1,109 @@
+use parking_lot::RwLock;
+use std::sync::Arc;
+
+use crate::class::ClassId;
+use crate::field::{Field, FieldDef};
+use crate::size::InstanceSize;
+use crate::ty::BuiltinType;
+use crate::utils::GrowableVec;
+use crate::vm::{FctId, FileId, ImplId, TraitId, VM};
+use crate::vtable::VTableBox;
+
+use dora_parser::interner::Name;
+use dora_parser::lexer::position::Position;
+
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct ModuleId(usize);
+
+impl ModuleId {
+    pub fn max() -> ModuleId {
+        ModuleId(usize::max_value())
+    }
+}
+
+impl From<ModuleId> for usize {
+    fn from(data: ModuleId) -> usize {
+        data.0
+    }
+}
+
+impl From<usize> for ModuleId {
+    fn from(data: usize) -> ModuleId {
+        ModuleId(data)
+    }
+}
+
+impl GrowableVec<RwLock<Module>> {
+    pub fn idx(&self, index: ModuleId) -> Arc<RwLock<Module>> {
+        self.idx_usize(index.0)
+    }
+}
+
+pub static DISPLAY_SIZE: usize = 6;
+
+#[derive(Debug)]
+pub struct Module {
+    pub id: ModuleId,
+    pub file: FileId,
+    pub pos: Position,
+    pub name: Name,
+    pub ty: BuiltinType,
+    pub parent_class: Option<ClassId>,
+    pub internal: bool,
+    pub internal_resolved: bool,
+    pub has_constructor: bool,
+
+    pub constructor: Option<FctId>,
+    pub fields: Vec<Field>,
+    pub methods: Vec<FctId>,
+    pub virtual_fcts: Vec<FctId>,
```

Comment:
Do we really need `virtual_fcts`, `traits` and `impls` in a module? Maybe we've discussed that before already though.

Reply:
I will check, I would have removed the unnecessary things after the next phase anyway, I just wasn't sure what I would actually need before implementing it.

Looking at it right now:
- trait: Yes, modules should be able to implement traits.
- impl: Unneeded, probably.
- virtual_fcts: No idea, does the Vec include virtual funs inherited from the parent class?

Reply:
Removed `impl`!